### PR TITLE
Storage/fix Module Name in azfile, azdatalake, azblob

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

**Review by vibhansa-msft — APPROVED** [review]

File: sdk/storage/azblob/service/client_test.go [hunk 1]
```diff
@@ -314,6 +316,41 @@ func (s *ServiceRecordedTestsSuite) TestSetPropertiesSetCORSMultiple() {
 	_require.NoError(err)
 }
 
+type userAgentTest struct{}
+
+func (u userAgentTest) Do(req *policy.Request) (*http.Response, error) {
+	const userAgentHeader = "User-Agent"
+
+	currentUserAgentHeader := map[string][]string(req.Raw().Header)[userAgentHeader]
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/service/client_test.go:324]: You can use `req.Raw().Header.Get()` method here. Something like this, https://github.com/Azure/azure-storage-fuse/blob/feature/track2Migration/component/azstorage/utils.go#L380-L382

File: sdk/storage/azblob/service/client_test.go [hunk 2]
```diff
@@ -314,6 +316,41 @@ func (s *ServiceRecordedTestsSuite) TestSetPropertiesSetCORSMultiple() {
...
+	if !strings.HasPrefix(currentUserAgentHeader[0], "azsdk-go-azblob/"+exported.ModuleVersion) {
+		return nil, fmt.Errorf(currentUserAgentHeader[0] + " user agent doesn't match expected agent: azsdk-go-azdatalake/v1.2.0 (go1.19.3; Windows_NT)")
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/service/client_test.go:326]:
```suggestion
		return nil, fmt.Errorf(currentUserAgentHeader[0] + " user agent doesn't match expected agent: azsdk-go-azblob/"+exported.ModuleVersion)
```

File: sdk/storage/azdatalake/file/client_test.go [hunk 3]
```diff
@@ -82,12 +86,46 @@ type UnrecordedTestSuite struct {
...
-//	testcommon.ValidateErrorCode(_require, err, datalakeerror.PathNotFound)
-//}
+//		testcommon.ValidateErrorCode(_require, err, datalakeerror.PathNotFound)
+//	}
+type userAgentTest struct{}
+
+func (u userAgentTest) Do(req *policy.Request) (*http.Response, error) {
+	const userAgentHeader = "User-Agent"
+
+	currentUserAgentHeader := map[string][]string(req.Raw().Header)[userAgentHeader]
```
Comment (souravgupta-msft) [inline sdk/storage/azdatalake/file/client_test.go:100]: Same as previous. `req.Raw().Header.Get()` method can be used instead.

File: sdk/storage/azfile/service/client_test.go [hunk 4]
```diff
@@ -181,6 +185,42 @@ func (s *ServiceRecordedTestsSuite) TestAccountHourMetrics() {
 	_require.NoError(err)
 }
 
+type userAgentTest struct{}
+
+func (u userAgentTest) Do(req *policy.Request) (*http.Response, error) {
+	const userAgentHeader = "User-Agent"
+
+	currentUserAgentHeader := map[string][]string(req.Raw().Header)[userAgentHeader]
```
Comment (souravgupta-msft) [inline sdk/storage/azfile/service/client_test.go:193]: Same as previous

**Review by souravgupta-msft — APPROVED** [review]
Few comments, looks good overall :)

**Pull request merged**